File: Problem 1 Codes/1_chatgpt_sol1.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int T;
    cin >> T;

    while (T--) {
        int N;
        long long X;
        cin >> N >> X;

        vector<long long> A(N);
        unordered_set<long long> present;

        for (int i = 0; i < N; i++) {
            cin >> A[i];
            present.insert(A[i]);
        }

        long long start = A[0];
        long long low = min(start, X);
        long long high = max(start, X);

        bool possible = true;

        for (long long value = low; value <= high; value++) {
            if (present.find(value) == present.end()) {
                possible = false;
                break;
            }
        }

        cout << (possible ? "YES" : "NO") << '\n';
    }

    return 0;
}

File: Problem 1 Codes/1_chatgpt_sol1.java
import java.io.*;
import java.util.*;

class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int t = Integer.parseInt(br.readLine().trim());

        while (t-- > 0) {
            StringTokenizer st = new StringTokenizer(br.readLine());

            int n = Integer.parseInt(st.nextToken());
            long x = Long.parseLong(st.nextToken());

            long[] a = new long[n];
            HashSet<Long> present = new HashSet<>();

            st = new StringTokenizer(br.readLine());

            for (int i = 0; i < n; i++) {
                a[i] = Long.parseLong(st.nextToken());
                present.add(a[i]);
            }

            long low = Math.min(a[0], x);
            long high = Math.max(a[0], x);

            boolean possible = true;

            for (long value = low; value <= high; value++) {
                if (!present.contains(value)) {
                    possible = false;
                    break;
                }
            }

            System.out.println(possible ? "YES" : "NO");
        }
    }
}